darwin/torch/dataset.py

```diff
@@ -141,96 +240,190 @@ def __getitem__(self, index: int):
+
+        Notes
+        -----
+        The return value is a dict with the following fields:
+            image_id : int
+                Index of the image inside the dataset
+            annotations : list[Dict]
+                List of annotations, where each annotation is a dict with:
+                iscrowd : int
+                    Flag to denote where the annotation more than one instance (can be 0 or 1)
+                category_id : int
+                    The single label of the image selected.
+                segmentation : ndarray(1,)
+                    Array of points [x,y,x,y,x,y ...] composing the polygon enclosing the object
+                bbox : ndarray(1,)
+                    Coordinates of the bounding box enclosing the instance as [x, y, w, h]
+                area : float
+                    Area of the polygon
+        """
         with self.annotations_path[index].open() as f:
-            annotation = json.load(f)["annotations"]
+            annotations = json.load(f)["annotations"]
 
         # Filter out unused classes
-        annotation = [a for a in annotation if a["name"] in self.classes]
+        annotations = [a for a in annotations if a["name"] in self.classes]
 
         target = []
-        for obj in annotation:
-            new_obj = {"image_id": index,
-                       "iscrowd": 0,
-                       "category_id": self.classes.index(obj["name"]),
-                       "segmentation": [convert_polygon_to_sequence(obj["polygon"]["path"])]}
-
-            seg = np.array(new_obj["segmentation"][0])
-            xcoords = seg[0::2]
-            ycoords = seg[1::2]
-            x = np.max((0, np.min(xcoords) - 1))
-            y = np.max((0, np.min(ycoords) - 1))
-            w = (np.max(xcoords) - x) + 1
-            h = (np.max(ycoords) - y) + 1
-            new_obj["bbox"] = [x, y, w, h]
-
-            poly_area = polygon_area(xcoords, ycoords)
+        for annotation in annotations:
+            assert "name" in annotation
+            assert "polygon" in annotation
+            # Extract the sequence of coordinates from the polygon annotation
+            sequence = convert_polygon_to_sequence(annotation["polygon"]["path"])
+            if len(sequence) / 2 < 3:
+                # Discard polygons with less than three points
+                continue
+            # Compute the bbox of the polygon
+            x_coords = sequence[0::2]
+            y_coords = sequence[1::2]
+            x = np.max((0, np.min(x_coords) - 1))
+            y = np.max((0, np.min(y_coords) - 1))
+            w = (np.max(x_coords) - x) + 1
+            h = (np.max(y_coords) - y) + 1
+            # Compute the area of the polygon
+            poly_area = polygon_area(x_coords, y_coords)
             bbox_area = w * h
             if poly_area > bbox_area:
                 raise ValueError(
                     f"polygon's area should be <= bbox's area. Failed {poly_area} <= {bbox_area}"
                 )
```

Comment:
ValueError seems a bit weird here, I see ValueError as the input to the function was of the correct type but the value doesn't fit in. like `div(a: int, b:int) , if b == 0 raise ValueError()`
But here it's the json we are parsing that might have invalid data, maybe a custom error would make sense?

Reply:
Fixed